core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -411,7 +412,7 @@ public boolean deleteFromAllStores(Object key, AccessMode mode) {
          boolean removed = false;
          for (CacheWriter w : writers) {
             if (mode.canPerform(configMap.get(w))) {
-               removed |= w.delete(key);
+               removed |= w.delete(MarshalledValue.unwrap(key));
```

Comment:
I guess this is one more place where a key-agnostic `CacheWriter` could have avoided a deserialization + serialization by dealing with the `MarshalledValue` directly. I'm starting to have some doubts about this change...